run.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+#
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# A convenience script to help developers easily set up emulator, finapp server
+# and workload.
+
+exit_with_help_msg() {
+  echo "Invalid command specified. Usage examples:
+  $ bash run.sh emulator
+  $ bash run.sh server java --spanner_project_id=test-project \
+      --spanner_instance_id=test-instance --spanner_database_id=test-database
+  $ bash run.sh server jdbc --spanner_project_id=test-project \
+      --spanner_instance_id=test-instance --spanner_database_id=test-database
+  $ bash run.sh workload \
+       --address-name localhost --port 8080 --thread-count 200
+  " >&2
+  exit 1
+}
+
+teardown_previous_emulator() {
+  # There doesn't seem to be a convenient gcloud command such as:
+  # gcloud emulators spanner stop. So we kill the processes on the ports.
+  echo "Tearing down any emulator processes running previously."
+  fuser -k 9010/tcp
+  fuser -k 9020/tcp
+}
+
+run_and_configure_emulator() {
+  teardown_previous_emulator
+  gcloud emulators spanner start &
+  if ! gcloud config configurations create emulator; then
+    # If this isn't the first time this script is being executed,
+    # create emulator will fail. Just activate the emulator in that case.
+    gcloud config configurations activate emulator
+  fi
+  gcloud config set auth/disable_credentials true
+  gcloud config set project test-project
+  gcloud config set api_endpoint_overrides/spanner http://localhost:9020/
+  gcloud spanner instances create test-instance \
+    --config=emulator-config --description="Test Instance" --nodes=1
+  gcloud config set spanner/instance test-instance
+  gcloud spanner databases create test-database \
+    --ddl-file server/src/main/java/com/google/finapp/schema.sdl
+  export SPANNER_EMULATOR_HOST="localhost:9010"
+}
+
+run_server_java() {
+  mvn clean compile assembly:single -pl org.example:server
+  java -jar server/target/server-1.0-SNAPSHOT-jar-with-dependencies.jar $@
```

Comment:
Any way to get the file name from the last command?

Reply:
Hm not quite sure. I think this can be replaced with `mvn exec java className` to make it slightly less brittle. I didn't want to deviate much from the existing instructions and just wrap them up in the script for convenience. We can perhaps revisit this if  this ends up causing maintenance burden.